VDA-Name-Registry/contracts/v2/NameRegistryV2.sol

```diff
@@ -0,0 +1,548 @@
+
+        AppMetaDataItem[] storage appData = _didAppMetaData[did][appName];
+        uint len = metadata.length;
+        for (uint i; i < len;) {
+            appData.push(metadata[i]);
+            unchecked {
+                ++i;
+            }
+        }
+    }
+    
+    /**
+     * @dev See {INameRegistryV2}
+     */
+    function registerApp(
+        address did,
+        string calldata ownerName,
+        string calldata appName,
+        AppMetaDataItem[] calldata metadata,
+        bytes calldata requestSignature,
+        bytes calldata requestProof
+    ) external virtual override {
+        string memory _ownerName;
+        string memory _appName;
+
+        if (!_isAppRegisterEnabled) {
+            revert AppRegisterNotEnabled();
+        }
+
+        // Check input values and verify request
+        {
+            _ownerName = _validateOwnerName(did, ownerName);
+            _appName = _validateAppName(did, appName);
+
+            bytes memory params = _getAppMetaDataPacked(metadata);
+            params = abi.encodePacked(did, ownerName, appName, params);
+
+            verifyRequest(did, params, requestSignature, requestProof);
+        }
+        
+        _receiveAppFee(tx.origin);
+
+        _registerAppInfo(did, _ownerName, _appName, metadata);
+
+        emit RegisterApp(did, ownerName, appName, metadata);
+    }
+
+    /**
+     * @notice Check the owner name & app name are registered to the DID
+     * @dev Used in `deRegisterApp()` and `updateApp()` functions
+     * @param did DID
+     * @param ownerName Owner name - lowercased
+     * @param appName App name - lowercased
+     */
+    function _validateExistingApp(
+        address did,
+        string memory ownerName,
+        string memory appName
+    ) internal view {
+
+        if (_ownerNameInfo[ownerName].DID != did) {
+            revert AppNotFound(true, false);
+        }
+
+        if (!_didApps[did].contains(appName)) {
+            revert AppNotFound(false, true);
+        }
+    }
+
+    /**
+     * @dev See {INameRegistryV2}
+     */
+    function deRegisterApp(
+        address did, 
+        string calldata ownerName, 
+        string calldata appName,
+        bytes calldata requestSignature,
+        bytes calldata requestProof
+    ) external virtual override {
+        string memory _ownerName;
+        string memory _appName;
+
+        _ownerName = ownerName.lower();
+        _appName = appName.lower();
+
+        
+        // verify request
+        {
+            _validateExistingApp(did, _ownerName, _appName);
+
+            bytes memory params = abi.encodePacked(did, ownerName, appName);
+            verifyRequest(did, params, requestSignature, requestProof);
+        }
+
+        // de-register
+        delete _didOwnerName[did];
+        delete _ownerNameInfo[_ownerName];
+
+        _didApps[did].remove(_appName);
+        delete _didAppMetaData[did][_appName];
+
+        emit DeRegisterApp(did, _ownerName, _appName);
```

Comment:
`emit deregisterApp`

Reply:
All the other event names are started with the uppercase letter.
For example, the event for the `registerApp()` function is `RegisterApp`.
How about to name the event for the `deregisterApp()` function as `Deregisterapp`?

Reply:
Yes `DeregisterApp`

Reply:
Resolved.